main/lsp/LocalVarSaver.cc

```diff
@@ -8,28 +8,20 @@ namespace sorbet::realmain::lsp {
 ast::TreePtr LocalVarSaver::postTransformLocal(core::Context ctx, ast::TreePtr tree) {
     auto &local = ast::cast_tree_nonnull<ast::Local>(tree);
 
-    core::SymbolRef owner;
+    core::MethodRef enclosingMethod;
     if (ctx.owner.data(ctx)->isMethod()) {
-        owner = ctx.owner;
+        enclosingMethod = ctx.owner.asMethodRef();
     } else if (ctx.owner == core::Symbols::root()) {
-        owner = ctx.state.lookupStaticInitForFile(core::Loc(ctx.file, local.loc));
+        enclosingMethod = ctx.state.lookupStaticInitForFile(core::Loc(ctx.file, local.loc));
     } else {
-        owner = ctx.state.lookupStaticInitForClass(ctx.owner.asClassOrModuleRef());
+        enclosingMethod = ctx.state.lookupStaticInitForClass(ctx.owner.asClassOrModuleRef());
     }
 
-    bool lspQueryMatch = ctx.state.lspQuery.matchesVar(owner, local.localVariable);
+    bool lspQueryMatch = ctx.state.lspQuery.matchesVar(enclosingMethod, local.localVariable);
     if (lspQueryMatch) {
         // No need for type information; this is for a reference request.
         // Let the default constructor make tp.type an empty shared_ptr and tp.origins an empty vector
         core::TypeAndOrigins tp;
-
-        auto enclosingMethod = ctx.owner;
```

Comment:
This whole block of code reimplemented what was occurring a few lines above, so I removed it.